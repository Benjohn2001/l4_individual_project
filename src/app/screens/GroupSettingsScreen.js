import React from 'react';
import { Feather } from '@expo/vector-icons';
import { View, Image, TextInput, Text, TouchableOpacity, SafeAreaView, FlatList, ActivityIndicator } from 'react-native';
import AppButtonPurple from '../components/AppButtonPurple'
import AppButtonLight from '../components/AppButtonLight';
import TwoButtonsSide from '../components/TwoButtonsSide';
import { COLOURS } from '../assets/colours';
import RowItem from '../components/RowItem';
import PressableIcon from '../components/PressableIcon';
import Modal from "react-native-modal";
import TwoButtonStack from '../components/TwoButtonStack';
import { deleteObject, getDownloadURL, getStorage } from 'firebase/storage';
import { getDatabase, ref, set, onValue, orderByChild, query, equalTo, get, push, remove, update} from 'firebase/database';
import SearchBar from "react-native-dynamic-search-bar";
import FriendsRowSearch from '../components/FriendsRowSearch';
import { auth } from '../../firebase';
import AddFromFriends from '../components/AddFromFriends';
import {ref as stref} from "firebase/storage";
import { Alert } from 'react-native';
import randomColor from "randomcolor";
import { useIsFocused } from '@react-navigation/native';

const GroupSettingsScreen = ({ route, navigation }) => {

    const item=route.params.item
    const name=item.val()["name"]
    const membRef=item.val()["membersRef"]
    const key=item.key

    const [searchVis, setSearchVis] = React.useState(false)

    const [isLeaveModalVisible, setLeaveModalVisible] = React.useState(false)
    const [isChangeModalVisible, setChangeModalVisible] = React.useState(false)
    const [spinVis, setSpinVis] = React.useState(false)
    const [unames, setUnames] = React.useState([])
    const [friends, setFriends] = React.useState([])
    const [friendsKeys, setFriendsKeys] = React.useState([])
    const [filtered, setFiltered] = React.useState([])
    const [membersKeys, setMembersKeys] = React.useState([])
    const [fetched, setFetched]= React.useState(false)
    const [nameNew, setNameNew] = React.useState("")
    const uid = auth.currentUser.uid

    React.useEffect(()=>{
        fetchData()
        getFriends()
    }, [searchVis])

    const fetchData = async () =>{
        setUnames([])
        setMembersKeys([])
        const unamesRef = query(ref(getDatabase(), "/users"), orderByChild("userName"))
        const data = await get(unamesRef)
        data.forEach(c => {
            if(c.val()["email"].toLowerCase() != auth.currentUser.email.toLowerCase()){
                setUnames(a => {return [...a , c]})
            }
        })
        const groupMembersRef = query(ref(getDatabase(), "/groupMembers/"+membRef))
        const dataMem = await get(groupMembersRef)
        for (const c in dataMem.val()) {
            setMembersKeys(a => {return [...a , dataMem.val()[c]["member"]]})
        }

    }

    const getFriends = async () =>{
        setFetched(false)
        setFriendsKeys([])
        setFriends([])
        const friendsRef = ref(getDatabase(), "/friends/"+uid)
        const dataFr = await get(friendsRef)
        for(const c in dataFr.val()){
            //console.log(membersKeys)
            //console.log(dataFr.val()[c]["user"])
            //console.log(!membersKeys.includes(dataFr.val()[c]["user"]))

            // trying to not show friends already in the group not working on first load, works after refresh
            //array of friendsKeys has all friends on first, then members removed after refresh??
            if(!membersKeys.includes(dataFr.val()[c]["user"])){
                setFriendsKeys(a =>{return [...a, dataFr.val()[c]["user"]]})
            }
            //console.log(c)
            //console.log(friendsKeys)
            //console.log(membersKeys )
        }

        for(const i in unames){
            if(friendsKeys.includes(unames[i].key)){
                setFriends(a => {return [...a , unames[i]]})
            }
        }
        setFetched(true)
    }

    const filterUname = (name) => {
        setFiltered([])
        friends.filter((str)=>{
                if(str.val()["userName"].toLowerCase().includes(name.toLowerCase())){
            
                    setFiltered(a => {return [...a , str]}) 
                }
        })
    }



    return (      
        <SafeAreaView className="flex-1 bg-primaryPurple" >
             {searchVis ?
                <>
                {fetched ?
                    <><View className="flex-row justify-between pt-14">
                        <View className="ml-auto">
                            <PressableIcon
                                onPress={() => {
                                    setSearchVis(false)
                                } }
                                icon="arrow-left"
                                size={40}
                                color="black" />
                        </View>
                        <Text className="font-bold ml-auto mr-auto text-3xl">
                            Add From Friends
                        </Text>
                    </View><View className="pt-10">
                            <SearchBar
                                className="bg-secondaryPurple h-12 w-11/12"
                                placeholder="Search for a username"
                                iconColor="6B4EFF"
                                spinnerVisibility={spinVis}
                                onChangeText={text => {
                                    setFiltered([]);
                                    if (text.length === 0) {
                                        setSpinVis(false);
                                    } else {
                                        setSpinVis(true);
                                    }
                                    filterUname(text);
                                } }

                                onClearPress={() => {
                                    filterUname('');
                                    setSpinVis(false);
                                } } />
                            <FlatList
                                data={filtered}
                                showsVerticalScrollIndicator={false}
                                renderItem={({ item }) => (
                                    <AddFromFriends
                                        title={item.val()["userName"]}
                                        onPress={async () => {
                                            const groupsRef = ref(getDatabase(), "/groups/" + item.key +"/" +key);
                                            const groupMembersRef = push(ref(getDatabase(), "/groupMembers/" + membRef));

                                            await update(groupsRef, {
                                                name: name,
                                                membersRef: membRef
                                            });
                                            await set(groupMembersRef, {
                                                member: item.key,
                                                colour: randomColor()
                                            });

                                            setSearchVis(false)
                                        } }
                                        avatar={item.val()["profilePic"]} />
                                )} />
                        </View></>
                :
                    <View className="justify-center items-center flex-1">
                        <ActivityIndicator size="large" color="#6B4EFF"  />
                        <Text className="text-center">Loading Friends</Text>
                    </View>
                }
            </>

            :
            <><View className="flex-row justify-between pt-14 mx-10">
                    <PressableIcon
                        onPress={() => {
                            navigation.goBack();
                        } }
                        icon="arrow-left"
                        size={40}
                        color="black" />
                    <Text className="font-bold ml-auto mr-auto text-3xl">
                        Group Settings
                    </Text>
                </View><Text className="font-bold text-2xl justify-start pt-10 mx-5 ">
                        Add to Group
                    </Text><View className='items-center'>
                        <RowItem
                            title="From Friends"
                            icon="users"
                            onPress={() => {
                                setSearchVis(true);
                            } }
                            color="black" />
                        <RowItem
                            title="Invite Friends"
                            icon="share-2"
                            onPress={() => {
                                alert("invite Friends");
                            } }
                            color="black" />
                    </View><Text className="font-bold text-2xl justify-start pt-10 mx-5 ">
                        Admin
                    </Text><View className='items-center'>
                        <RowItem
                            title="Change Group Name"
                            icon="edit-3"
                            onPress={() => {
                                setChangeModalVisible(true)
                            } }
                            color="black" />
                        <RowItem
                            title="Leave Group"
                            icon="trash-2"
                            onPress={() => {
                                setLeaveModalVisible(true);
                            } }
                            color="red" />
                    </View><Modal
                        isVisible={isLeaveModalVisible}
                        onBackdropPress={() => setLeaveModalVisible(false)}
                        className="items-center"
                    >
                        <View className=" w-full h-1/4 items-center py-3 px-3 bg-secondaryPurple rounded-2xl">
                            <Text className="text-2xl font-bold  pb-2">Leave Group</Text>
                            <Text className="text-l pb-2">Are you sure you want to leave this group?</Text>
                            <View className="pt-5">
                                <TwoButtonsSide
                                    title1="Cancel"
                                    onPress1={() => { setLeaveModalVisible(false); } }
                                    icon1="x-circle"
                                    color1="#C6C4FF"
                                    title2="Leave"
                                    onPress2={async () => {
                                        const groupsRef = query(ref(getDatabase(), "/groups/" + uid));
                                        const groupMembersRef = query(ref(getDatabase(), "/groupMembers/" + membRef));
                                        const dataGroups = await get(groupsRef);
                                        if (dataGroups.val() !== null) {
                                            dataGroups.forEach(c => {
                                                if (c.val()["membersRef"] == membRef) {
                                                    remove(ref(getDatabase(), "/groups/" + uid + "/" + c.key));
                                                }
                                            });
                                        }
                                        const dataMembers = await get(groupMembersRef);
                                        if (dataMembers.val() !== null) {
                                            dataMembers.forEach(c => {
                                                if (c.val()["member"] == uid) {
                                                    remove(ref(getDatabase(), "/groupMembers/" + membRef + "/" + c.key));
                                                }
                                            });
                                        }
                                        if(dataMembers.size===1){
                                            remove(ref(getDatabase(), "/locations/" + membRef ))
                                            remove(ref(getDatabase(), "/clockFace/" + membRef))
                                            
                                            const file = stref(getStorage(),"images/clockFaces/"+membRef)
                                            await getDownloadURL(file).then(deleteObject(stref(getStorage(),"images/clockFaces/"+membRef)))
                                            .catch(error =>{
                                                if(error.code === 'storage/object-not-found'){
                                                    console.log("no image in storage")
                                                }
                                            })
                                        }
                                        setLeaveModalVisible(false);
                                        navigation.navigate("HomeScreen");
                                    } }
                                    icon2="trash-2"
                                    color2="red" />
                            </View>

                        </View>


                    </Modal>
                    <Modal 
                        isVisible={isChangeModalVisible}
                        onBackdropPress={() => setChangeModalVisible(false)}
                        className="items-center"
                    >
                        <View className=" w-11/12 h-50 items-center py-3 px-3 bg-secondaryPurple rounded-2xl">
                            <Text className="text-2xl font-bold  pb-2">Change group name</Text>
                            <Text className="text-center text-gray-500 py-4">Enter a new group name</Text>
                            <TextInput 
                                className="bg-white my-5 w-full mx-3 h-12 rounded-md" 
                                placeholder="group name" 
                                underlineColorAndroid = "transparent"
                                cursorColor={COLOURS.darkerPurple}
                                maxLength={50}
                                value={nameNew}
                                onChangeText={(val) => setNameNew(val)}
                            />
                            <AppButtonPurple
                                title="Change Name"
                                onPress={async ()=>{
                                    for (const i in membersKeys) {
                                            await update(ref(getDatabase(), '/groups/' + membersKeys[i] + "/" + key ), {
                                                name: nameNew
                                            })
     
                                    }
                                
                                    setNameNew("")
                                    setChangeModalVisible(false)
                                    navigation.navigate("HomeScreen")
                                }}
                            />
                            
                        </View>
                    </Modal>
                    
                    
                    </>
            }

        </SafeAreaView>

    );
};

export default GroupSettingsScreen;