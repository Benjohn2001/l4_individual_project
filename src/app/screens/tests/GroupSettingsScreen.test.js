import renderer from "react-test-renderer";
import { render, screen, fireEvent } from "@testing-library/react-native";
import GroupSettingsScreen from "../GroupSettingsScreen";

jest.mock("firebase/database", () => ({
  getDatabase: jest.fn().mockReturnThis(),
  push: jest.fn().mockReturnThis(),
  ref: jest.fn().mockReturnThis(),
  set: jest.fn().mockReturnThis(),
  query: jest.fn().mockReturnThis(),
  onValue: jest.fn().mockReturnThis(),
}));

jest.mock("firebase/storage", () => ({
  getStorage: jest.fn().mockReturnThis(),
}));

const navigation = {
  navigate: jest.fn(),
  push: jest.fn(),
  goBack: jest.fn(),
};

jest.mock("firebase/app", () => ({
  auth: jest.fn().mockReturnThis(),
  initializeApp: jest.fn().mockReturnThis(),
}));

jest.mock("firebase/auth", () => ({
  auth: jest.fn().mockReturnThis(),
  getAuth: () => ({
    currentUser: jest.fn().mockReturnThis(),
    signOut: () => ({
      then: () => ({
        catch: jest.fn(),
      }),
    }),
  }),
}));

const route = {
  params: {
    item: {
      val: jest.fn().mockReturnThis(),
    },
  },
};

test("Check GroupSettingsScreen loading correctly", async () => {
  render(<GroupSettingsScreen navigation={navigation} route={route} />);

  const domTree = renderer
    .create(<GroupSettingsScreen route={route} />)
    .toJSON();
  expect(domTree).toMatchSnapshot();
});

test("Check GroupSettingsScreen navigating correctly", async () => {
  render(<GroupSettingsScreen navigation={navigation} route={route} />);
  fireEvent.press(screen.getByTestId("pressableIcon"));
  expect(navigation.goBack).toHaveBeenCalledTimes(1);
});

test("Check from friends", async () => {
  render(<GroupSettingsScreen navigation={navigation} route={route} />);
  fireEvent.press(screen.getByText("From Friends"));
});
test("Check invite friends", async () => {
  render(<GroupSettingsScreen navigation={navigation} route={route} />);
  fireEvent.press(screen.getByText("Invite Friends"));
});

test("Check leave group cancel", async () => {
  render(<GroupSettingsScreen navigation={navigation} route={route} />);
  fireEvent.press(screen.getByText("Leave Group"));
  fireEvent.press(screen.getByText("Cancel"));
});

test("Check leave group leave", async () => {
  render(<GroupSettingsScreen navigation={navigation} route={route} />);
  fireEvent.press(screen.getByText("Leave Group"));
  fireEvent.press(screen.getByText("Leave"));
});

test("Check change group name", async () => {
  render(<GroupSettingsScreen navigation={navigation} route={route} />);
  fireEvent.press(screen.getByText("Change Group Name"));
  fireEvent.changeText(screen.getByPlaceholderText("group name"), "abc");
  fireEvent.press(screen.getByText("Change Name"));
});
