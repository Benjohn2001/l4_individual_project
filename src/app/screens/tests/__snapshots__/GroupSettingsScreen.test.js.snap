// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check GroupSettingsScreen loading correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "flexBasis": "0%",
        "flexGrow": 1,
        "flexShrink": 1,
      },
      Object {
        "backgroundColor": "#C6C4FF",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
        },
        Object {
          "justifyContent": "space-between",
        },
        Object {
          "paddingTop": 56,
        },
        Object {
          "marginLeft": 40,
          "marginRight": 40,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
        ]
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        testID="pressableIcon"
      >
        <Text />
      </View>
    </View>
    <Text
      style={
        Array [
          Object {
            "fontWeight": "700",
          },
          Object {
            "marginLeft": "auto",
          },
          Object {
            "marginRight": "auto",
          },
          Object {
            "fontSize": 30,
            "lineHeight": 36,
          },
        ]
      }
    >
      Group Settings
    </Text>
  </View>
  <Text
    style={
      Array [
        Object {
          "fontWeight": "700",
        },
        Object {
          "fontSize": 24,
          "lineHeight": 32,
        },
        Object {
          "justifyContent": "flex-start",
        },
        Object {
          "paddingTop": 40,
        },
        Object {
          "marginLeft": 20,
          "marginRight": 20,
        },
      ]
    }
  >
    Add to Group
  </Text>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "paddingTop": 12,
          },
          Object {
            "width": "100%",
          },
          Object {
            "alignItems": "center",
          },
        ]
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#E7E7FF",
            "borderBottomLeftRadius": 16,
            "borderBottomRightRadius": 16,
            "borderTopLeftRadius": 16,
            "borderTopRightRadius": 16,
            "height": 56,
            "marginLeft": 6,
            "marginRight": 6,
            "opacity": 1,
            "paddingTop": 16,
            "width": "91.666667%",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "marginLeft": 20,
                },
              ]
            }
          >
            <Text />
          </View>
          <Text
            style={
              Array [
                Object {
                  "color": "#000",
                },
                Object {
                  "fontSize": 14,
                  "lineHeight": 20,
                },
                Object {
                  "textAlign": "center",
                },
                Object {
                  "marginLeft": 40,
                },
              ]
            }
          >
            From Friends
          </Text>
          <View
            style={
              Array [
                Object {
                  "marginLeft": "auto",
                },
                Object {
                  "marginRight": 20,
                },
              ]
            }
          >
            <Text />
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "paddingTop": 12,
          },
          Object {
            "width": "100%",
          },
          Object {
            "alignItems": "center",
          },
        ]
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#E7E7FF",
            "borderBottomLeftRadius": 16,
            "borderBottomRightRadius": 16,
            "borderTopLeftRadius": 16,
            "borderTopRightRadius": 16,
            "height": 56,
            "marginLeft": 6,
            "marginRight": 6,
            "opacity": 1,
            "paddingTop": 16,
            "width": "91.666667%",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "marginLeft": 20,
                },
              ]
            }
          >
            <Text />
          </View>
          <Text
            style={
              Array [
                Object {
                  "color": "#000",
                },
                Object {
                  "fontSize": 14,
                  "lineHeight": 20,
                },
                Object {
                  "textAlign": "center",
                },
                Object {
                  "marginLeft": 40,
                },
              ]
            }
          >
            Invite Friends
          </Text>
          <View
            style={
              Array [
                Object {
                  "marginLeft": "auto",
                },
                Object {
                  "marginRight": 20,
                },
              ]
            }
          >
            <Text />
          </View>
        </View>
      </View>
    </View>
  </View>
  <Text
    style={
      Array [
        Object {
          "fontWeight": "700",
        },
        Object {
          "fontSize": 24,
          "lineHeight": 32,
        },
        Object {
          "justifyContent": "flex-start",
        },
        Object {
          "paddingTop": 40,
        },
        Object {
          "marginLeft": 20,
          "marginRight": 20,
        },
      ]
    }
  >
    Admin
  </Text>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "paddingTop": 12,
          },
          Object {
            "width": "100%",
          },
          Object {
            "alignItems": "center",
          },
        ]
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#E7E7FF",
            "borderBottomLeftRadius": 16,
            "borderBottomRightRadius": 16,
            "borderTopLeftRadius": 16,
            "borderTopRightRadius": 16,
            "height": 56,
            "marginLeft": 6,
            "marginRight": 6,
            "opacity": 1,
            "paddingTop": 16,
            "width": "91.666667%",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "marginLeft": 20,
                },
              ]
            }
          >
            <Text />
          </View>
          <Text
            style={
              Array [
                Object {
                  "color": "#000",
                },
                Object {
                  "fontSize": 14,
                  "lineHeight": 20,
                },
                Object {
                  "textAlign": "center",
                },
                Object {
                  "marginLeft": 40,
                },
              ]
            }
          >
            Change Group Name
          </Text>
          <View
            style={
              Array [
                Object {
                  "marginLeft": "auto",
                },
                Object {
                  "marginRight": 20,
                },
              ]
            }
          >
            <Text />
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "paddingTop": 12,
          },
          Object {
            "width": "100%",
          },
          Object {
            "alignItems": "center",
          },
        ]
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#E7E7FF",
            "borderBottomLeftRadius": 16,
            "borderBottomRightRadius": 16,
            "borderTopLeftRadius": 16,
            "borderTopRightRadius": 16,
            "height": 56,
            "marginLeft": 6,
            "marginRight": 6,
            "opacity": 1,
            "paddingTop": 16,
            "width": "91.666667%",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "marginLeft": 20,
                },
              ]
            }
          >
            <Text />
          </View>
          <Text
            style={
              Array [
                Object {
                  "color": "#000",
                },
                Object {
                  "fontSize": 14,
                  "lineHeight": 20,
                },
                Object {
                  "textAlign": "center",
                },
                Object {
                  "marginLeft": 40,
                },
              ]
            }
          >
            Leave Group
          </Text>
          <View
            style={
              Array [
                Object {
                  "marginLeft": "auto",
                },
                Object {
                  "marginRight": 20,
                },
              ]
            }
          >
            <Text />
          </View>
        </View>
      </View>
    </View>
  </View>
  <Modal
    animationType="none"
    deviceHeight={null}
    deviceWidth={null}
    hardwareAccelerated={false}
    hideModalContentWhileAnimating={false}
    onBackdropPress={[Function]}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    onRequestClose={[Function]}
    panResponderThreshold={4}
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    statusBarTranslucent={false}
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
    transparent={true}
    visible={false}
  />
  <Modal
    animationType="none"
    deviceHeight={null}
    deviceWidth={null}
    hardwareAccelerated={false}
    hideModalContentWhileAnimating={false}
    onBackdropPress={[Function]}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    onRequestClose={[Function]}
    panResponderThreshold={4}
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    statusBarTranslucent={false}
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
    transparent={true}
    visible={false}
  />
</RCTSafeAreaView>
`;
